.opspec/build/op.yml

```diff
@@ -13,10 +13,15 @@ inputs:
     string:
       constraints:
         format: semver
+  alreadyPublished:
```

Comment:
you might consider conditionally calling this or not within the release op as opposed to making it a parameter since if you're calling build you generally would want to build or you just wouldn't call it.

Reply:
here's an idea of what I meant

```yaml
description: performs a release
name: release
inputs:
  dockerSocket:
    socket:
      description: docker socket to use as container runtime 
  gitBranch:
    string:
      constraints: { minLength: 1 }
      description: git branch the coverage should be associated with
      default: main
  github:
    object:
      constraints:
        properties:
          accessToken:
            minLength: 1
            type: string
            writeOnly: true
          username:
            minLength: 1
            type: string
        required: [accessToken, username]
      description: configuration required to interact w/ github
  HOME:
    dir:
      description: Home directory of caller; used to access go modules
  version:
    string:
      constraints: { format: semver }
      description: version of opctl being released
run:
  serial:
    - op:
        ref: $(../changelog/get-latest-version)
        outputs:
          latestVersion:
    - op:
        ref: github.com/opspec-pkgs/base64.encode#1.1.0
        inputs:
          rawValue: $(github.username):$(github.accessToken)
        outputs:
          encodedValue: $(b64GithubAuth)
    - container:
        image: { ref: alpine:3.20 }
        envVars:
          version: $(version)
        cmd:
          - sh
          - -c
          - /check.sh
        sockets:
          /var/run/docker.sock: $(dockerSocket)
        files:
          /alreadyPublished: $(alreadyPublished)
          /check.sh:
          /root/.docker/config.json:
            auths:
              ghcr.io:
                auth: $(b64GithubAuth)
    - if:
        - eq: [$(alreadyPublished), false]
      serial:
        - op:
            ref: $(../build)
            inputs:
              dockerSocket:
              gitBranch:
              HOME:
              version: $(latestVersion)
              alreadyPublished:
        - parallel:
            - op:
                ref: $(./to-ghcr)
                inputs:
                  version: $(latestVersion)
                  alreadyPublished: $(alreadyPublished)
                  encodedGithubAuth: $(b64GithubAuth)
                  dockerSocket:
            - op:
                ref: $(./to-github)
                inputs:
                  github:
                  version: $(latestVersion)
                  alreadyPublished: $(alreadyPublished)
```

Reply:
done! Didn't realize you could do that